### PR TITLE
Removed legacy log files from 2.2 and backwards

### PR DESCRIPTION
**Pull request by klaren**
The old legacy format for log files that had a version header in the file can be removed since 2.2 is no longer supported. The implication of this is that 3.3 can only read log files back to 2.3 during migration.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/VersionAwareLogEntryReader.java [hunk 1]
```diff
@@ -70,24 +69,12 @@ public LogEntry readLogEntry( SOURCE channel ) throws IOException
...
-                        // if the read type is negative than it is actually the log entry version
-                        // so we need to read an extra byte which will contain the type
-                        versionCode = typeCode;
-                        typeCode = channel.get();
-                    }
-                }
-                catch ( ReadPastEndException e )
-                {   // Make these exceptions slip by straight out to the outer handler
-                    throw e;
+                    return null;
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/VersionAwareLogEntryReader.java:75]: Did this method previously also explicitly return `null` on `versionCode` `0`? I can't see it straight away by looking at the existing code.

**Review by tinwelint — COMMENTED** [review]
Nice to get rid of all those old things, well done.

I have a question about a zero version code in `VersionAwareLogEntryReader`, otherwise things look good methinks.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/VersionAwareLogEntryReader.java (hunk 1 above)
Comment (klaren) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/VersionAwareLogEntryReader.java:75]: Not explicitly. The zero-version mapped to the old `LogEntryParsersV2_0` which mapped zero-type to `null`. This seems to be needed according to the test `shouldParseStreamOfZerosAsEmptyLogEntries()` in `VersionAwareLogEntryReaderTest`.

Comment (tinwelint) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/VersionAwareLogEntryReader.java:75]: OK cheers

**Review by tinwelint — APPROVED** [review]

**Pull request merged**